lib/process/process.go

```diff
@@ -866,22 +866,45 @@ func (p *Process) ReportReadiness(w http.ResponseWriter, r *http.Request) {
 // ReportHealth is HTTP check that reports that the system is healthy
 // if it can successfully connect to the storage backend
 func (p *Process) ReportHealth(w http.ResponseWriter, r *http.Request) {
-	_, err := p.backend.GetAccounts()
-	var statusCode int
-	var info string
-	if err == nil {
-		statusCode = http.StatusOK
-		info = "service is up and running"
-	} else {
-		p.Errorf("Error: %v.", trace.DebugReport(err))
-		statusCode = http.StatusServiceUnavailable
-		info = err.Error()
+	log := p.WithField(trace.Component, "healthz")
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	healthCh := make(chan error, 1)
+	go func() {
+		started := time.Now()
+		_, err := p.backend.GetAccounts()
+		elapsed := time.Now().Sub(started)
+		if elapsed > 25*time.Millisecond {
```

Comment:
I'm curious how you landed on this number.

Reply:
I ran a perf test on one of my clusters, where the slowest request took about 15 ms, so I configured this with a small margin of head room, since it's only used in logging, and I'd like to see it trigger the log if things are only a little slow.

Ideally, we would have embedded metrics, and I would be able to scrape the histogram over a prometheus endpoint for all etcd queries, but I think wiring in metrics should be a bigger undertaking.

I'll add a comment before merging, so this is clearer when it comes back up.